Userland/Libraries/LibProtocol/Request.cpp

```diff
@@ -140,7 +140,7 @@ void Request::set_up_internal_stream_data(DataReceived on_data_available)
 
         do {
             auto result = m_internal_stream_data->read_stream->read_some({ buffer, buffer_size });
-            if (result.is_error() && (!result.error().is_errno() || (result.error().is_errno() && result.error().code() != EINTR)))
+            if (result.is_error() && (!result.error().is_errno() || (result.error().is_errno() && (result.error().code() != EINTR && result.error().code() != EAGAIN))))
```

Comment:
This change fixes https://github.com/SerenityOS/serenity/pull/24423#discussion_r1611748985, but I'm not 100% convinced this is the best way to handle this error.  It seems like this could thrash the CPU if `read_some()` keeps returning `EAGAIN`, I haven't actually been able to find anywhere that this causes any real performance degradation though.

Do you have any thoughts on this approach @alimpfard?

Reply:
This seems fine, it's a tad odd for the notifier to say there's data available, but for the actual read to not succeed (this isn't a shared resource, so no racing here) though.